pulsar-common/src/main/java/org/apache/pulsar/common/policies/data/Policies.java

```diff
@@ -43,13 +45,19 @@
     public Set<String> replication_clusters = Sets.newHashSet();
     public BundlesData bundles;
     @SuppressWarnings("checkstyle:MemberName")
-    public Map<BacklogQuota.BacklogQuotaType, BacklogQuota> backlog_quota_map = Maps.newHashMap();
+    @JsonDeserialize(as = ConcurrentHashMap.class)
```

Comment:
I'd rather not follow this approach because it leaves out other fields (that are also non-thread-safe) and other classes that will have the same issue.

Instead, we should try to find a way to the handling of modifications to these policies is synchronized.

Reply:
@merlimat That's true that there is this uncomfortable feeling. :) However, it seems that this ConcurrentHashMap based solution makes a lot of sense in this context. One of the benefits is that the change doesn't require huge architectural changes and the risk that it impacts performance is very minor.

> Instead, we should try to find a way to the handling of modifications to these policies is synchronized.

Does that help? Wouldn't it mean that reads are also synchronized? Wouldn't that impact performance? Perhaps I misunderstood what you meant with "synchronized".

Reply:
> Wouldn't it mean that reads are also synchronized?

Not necessarily, we could enforce to always clone the POJO before updating them, in copy-on-write fashion.

Reply:
@merlimat  Yes, makes sense. There were some ideas in that direction in https://github.com/apache/pulsar/issues/9711#issuecomment-793770462 .

Reply:
I will create a new PR and use orderedExecutor to make it thread safe